### ISSUE TITLE
Can't connect to hue bridge

### ISSUE THREAD

**Issue by manueligno78** [ebaauw/homebridge-hue#51]
Im trying to test homebridge-hue, but at the start of homebridge i'm getting this log message:
```
[2017-01-18 23:10:37] [Hue] Philips hue: hue bridge request #108: post / {"devicetype":"homebridge-hue#Mac-mini-di-Emanuele-7.local"}
[2017-01-18 23:10:37] [Hue] Philips hue: hue bridge error 7: invalid value,  homebridge-hue#Mac-mini-di-Emanuele-7.local, for parameter, devicetype
[2017-01-18 23:10:37] [Hue] Philips hue: press link button on the bridge to create a user
```

Even if I press the link button, I get the same message.

This is my config.json:

```
{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
    "platforms": [
    {
      "platform": "Hue",
      "name": "Hue",
      "host": "192.168.1.137",
      "users": {
      },
      "heartrate": 5,
      "timeout": 5,
      "lights": false,
      "philipslights": false,
      "ct": false,
      "groups": false,
      "group0": true,
      "rooms": false,
      "sensors": true,
      "excludeSensorTypes": [],
      "schedules": false,
      "rules": false
    }
    ],
    "accessories": [
    ]
}
```

**Comment by manueligno78** [comment]
This is my log:
```
[2017-01-18 23:24:22] Loaded plugin: homebridge-http
[2017-01-18 23:24:22] Registering accessory 'homebridge-http.Http'
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loaded plugin: homebridge-hue
[2017-01-18 23:24:22] Registering platform 'homebridge-hue.Hue'
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loaded plugin: homebridge-server
[2017-01-18 23:24:22] Registering platform 'homebridge-server.Server'
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loaded plugin: homebridge-sky-plus-hd
[2017-01-18 23:24:22] Registering accessory 'homebridge-sky-plus-hd.SkyPlusHDBox'
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loaded plugin: homebridge-wol
[2017-01-18 23:24:22] Registering accessory 'homebridge-wol.Computer'
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loaded config.json with 0 accessories and 1 platforms.
[2017-01-18 23:24:22] ---
[2017-01-18 23:24:22] Loading 1 platforms...
[2017-01-18 23:24:22] [Hue] Initializing Hue platform...
[2017-01-18 23:24:22] [Hue] homebridge-hue v0.1.12, node v5.1.0, homebridge v2.1
[2017-01-18 23:24:22] [Hue] contacting meethue portal
[2017-01-18 23:24:22] Loading 0 accessories...
[2017-01-18 23:24:23] [Hue] probing bridge at 192.168.1.137
[2017-01-18 23:24:23] [Hue] 192.168.1.137: hue bridge request #1: get /config
[2017-01-18 23:24:23] [Hue] 192.168.1.137: hue bridge request #1: ok
[2017-01-18 23:24:23] [Hue] Philips hue: BSB002 bridge, api v1.16.0
[2017-01-18 23:24:23] [Hue] Philips hue: hue bridge request #2: post / {"devicetype":"homebridge-hue#Mac-mini-di-Emanuele-7.local"}
[2017-01-18 23:24:23] [Hue] Philips hue: hue bridge request #2: post / {"devicetype":"homebridge-hue#Mac-mini-di-Emanuele-7.local"}
[2017-01-18 23:24:23] [Hue] Philips hue: hue bridge error 7: invalid value,  homebridge-hue#Mac-mini-di-Emanuele-7.local, for parameter, devicetype
[2017-01-18 23:24:23] [Hue] Philips hue: press link button on the bridge to create a user
[2017-01-18 23:24:28] [Hue] Philips hue: hue bridge request #3: post / {"devicetype":"homebridge-hue#Mac-mini-di-Emanuele-7.local"}
[2017-01-18 23:24:28] [Hue] Philips hue: hue bridge request #3: post / {"devicetype":"homebridge-hue#Mac-mini-di-Emanuele-7.local"}
[2017-01-18 23:24:28] [Hue] Philips hue: hue bridge error 7: invalid value,  homebridge-hue#Mac-mini-di-Emanuele-7.local, for parameter, devicetype
[2017-01-18 23:24:28] [Hue] Philips hue: press link button on the bridge to create a user
```

**Comment by GatoPharaoh** [comment]
try removing `"users": {  },` from your config.json

**Comment by ebaauw** [comment]
homebridge-hue tries to create a bridge username, using the name of your server your server to construct a `devicetype`.  That fails, I think, because the `devicetype` is too long.

As a workaround you could:
- Create a username manually (see [https://developers.meethue.com/documentation/getting-started](https://developers.meethue.com/documentation/getting-started), or
- Change your server name to something shorter.

Structurally, I need to check how long the devicetype can be and make homebridge-hue truncate the server name if needed.

**Comment by ebaauw** [comment]
Fixed in v0.1.13.

**Comment by manueligno78** [comment]
test passed :)
